src/store/index.ts

```diff
@@ -0,0 +1,53 @@
+import { Store as ReduxStore, createStore, applyMiddleware, AnyAction } from 'redux';
+
+import { State } from '~/shared/state';
+import { IPullRequestManager } from '../github/interface';
+
+import reducer from './reducers';
+import Logger from '../common/logger';
+
+import newPRMiddleware from './middleware/newPR';
+import { Disposable, Event } from 'vscode';
+import { init } from './actions';
+
+export class Store {
```

Comment:
is this meant to ultimately store all of the extension's state? what's the responsibility of this class?

Centralizing state is fine, but I'm concerned that we now have duplicate sources of truth for some of the state - ex. the PullRequestManager is also tracking `githubRepositories`